src/tags.ts

```diff
@@ -68,30 +83,32 @@ export function applyTags(construct: IConstruct, ctx: TagsBase): void {
   if (ctx.data?.isPublic && ctx.classification !== SecurityClassification.Unclassified) {
     throw new Error('Only unclassified constructs can be made public');
   }
-  const buildInfo = getGitBuildInfo();
+
+  const buildInfo = ctx.commitInfo ? getGitBuildInfo() : undefined;
```

Comment:
this change is changing existing behaviour... should this be flipped around to reduce upgrade toil for existing consumers?